src/main/java/org/broadinstitute/hellbender/metrics/MultiLevelCollectorReducible.java

```diff
@@ -0,0 +1,428 @@
+                }
+                collector = makeUnknownCollector();
+                collectors.put(key, collector);
+            }
+            collector.acceptRecord(args);
+        }
+
+        /** Call finish on each PerUnitMetricCollector in this Aggregate Collector */
+        public void finish() {
+            for(final UNIT_COLLECTOR collector : collectors.values()) {
+                collector.finish();
+            }
+        }
+
+        /**
+         * Combine/merge the source distributor into this distributor by merging the collector maps.
+         * @param source source Distributor
+         * @param reMap map entry merge (reMap) function for combining distributor map entries. This
+         *              is usually the combineUnit method of MultiLevelCollectorReducible.
+         */
+        public void combine(Distributor source, BiFunction<UNIT_COLLECTOR, UNIT_COLLECTOR, UNIT_COLLECTOR> reMap) {
+            collectors.keySet().forEach(k -> collectors.merge(k, source.collectors.get(k), reMap));
+        }
+
+        /** Add all records to the MetricsFile passed in, this will happen in the order they were
+         * found in the input ReadGroup records */
+        public void addToFile(final MetricsFile<METRIC_TYPE, HISTOGRAM_KEY> file) {
+            for(final UNIT_COLLECTOR collector : collectors.values()) {
+                collector.addMetricsToFile(file);
+            }
+        }
+    }
+
+    /** A dummy Distributor to handle the ALL_READS accumulation level.  No distribution is required
+     * since there should only ever be one PerUnitMetricCollector for ALL_READS.
+     */
+    private class AllReadsDistributor extends Distributor implements Serializable {
+
+        private static final long serialVersionUID = 1L;
+
+        public AllReadsDistributor(final List<SAMReadGroupRecord> rgRecs) {
+            // NOTE: since we're passing an empty list to the super class constructor,
+            // no collector will actually be added to the collector map; instead this
+            // class assumes a hardcoded allReadsCollector in all of it's other methods
+            super(new ArrayList<>());
+            makeCollector(null);
+        }
+
+        @Override
+        protected String getKey(SAMReadGroupRecord rg) {
+            return null;
+        }
+
+        public void acceptRecord(final ARGTYPE args, final SAMReadGroupRecord rg) {
+            allReadCollector.acceptRecord(args);
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeCollector(final SAMReadGroupRecord rg) {
+            allReadCollector = makeAllReadCollector();
+            return allReadCollector;
+        }
+
+        @Override
+        protected UNIT_COLLECTOR makeUnknownCollector() {
+            throw new UnsupportedOperationException("Should not happen");
+        }
+
+        @Override
+        public void finish() {
+            allReadCollector.finish();
```

Comment:
Yes, the all reads collector is a special case. However as part of the change to static inner classes, I changed the all reads distributor to use the same collector list the other distributors use, so this method overload is no longer necessary at all.